microcosm_pubsub/daemon.py

```diff
@@ -9,18 +9,30 @@
 
 class ConsumerDaemon(Daemon):
 
+    def __init__(self):
+        super(ConsumerDaemon, self).__init__()
+        self.bound_handlers = None
+
     def make_arg_parser(self):
         parser = super(ConsumerDaemon, self).make_arg_parser()
         parser.add_argument("--sqs-queue-url")
         return parser
 
+    def create_object_graph_components(self, graph):
+        super(ConsumerDaemon, self).create_object_graph_components(graph)
+        self.bound_handlers = graph.sqs_message_handler_registry.compute_bound_handlers(
```

Comment:
It's possible to have multiple daemons in the same code base; in this case, the handler registry could have more than one entry. We solve this by determining which handlers are part of the daemon's object graph and restricting access to just those handlers.